docs/source/explanation/additional_features/auto_configuration.rst

```diff
@@ -1,4 +1,57 @@
 Auto-configuration
 ==================
 
-Coming soon...
\ No newline at end of file
+Auto-configuration for a deep learning framework means the automatic finding of the most appropriate settings for the training parameters, based on the dataset and the specific task at hand.
+Auto-configuration can help to save time, it eases the process of interaction with OTX CLI and gives a better baseline for the given dataset.
+
+At this end, we developed a simple auto-configuration functionality to ease the process of training and validation utilizing our framework.
+Basically, to start the training and obtain a good baseline with the best trade-off between accuracy and speed we need to pass only a dataset in the right format without specifying anything else:
+
+.. code-block::
+
+    $ otx train --train-data-root <path_to_data_root>
+
+.. note::
+
+    OTX supports also ``otx build`` mode with the auto-configuration feature. We can build OTX workspace with the following CLI command:
+
+    .. code-block::
+
+        $ otx build --train-data-root <path_to_data_root>
+
+Moreover, our dataset can have no train/val splits at all. The Datumaro manager integrated into OTX will handle it on its own.
+It will recognize the task by analyzing the dataset and if there is no splits for the validation - Datumaro will do a random auto-split, memorizing this split to further use it with ``otx eval`` and ``otx optimize``.
```

Comment:
One small suggestion, 

Since `otx build` will makes the `train` and `val` datasets, users can't directly use these datasets to `otx eval` which is required `test` dataset. In the case of `otx optimize`,there is no problem. 

So, how about converting like below?

It will recognize the task by analyzing the dataset and if there is no splits for the validation - Datumaro will do a random auto-split, memorizing this split to further use it with ``otx optimize``.

Reply:
Okay, maybe this is better, I changed it.